### ISSUE TITLE
Incorrect PlayerState object remapping/id syncing

### ISSUE THREAD

**Issue by envision3d** [FlaxEngine/FlaxEngine#1607]
**Issue description:**

Running the code below (starting at SRPCStep1()) results in the logs that follow (thread).  Also included in the logs are potentially relevant remapping logs.  It would seem that some objects are somehow referencing old ids?

Not shown here, but in my project I have some scripts that reference PlayerState.PlayerPawn in the update method, and when the issue described here occurs, PlayerState.PlayerPawn is null.

I've reproduced this in my project as well as the ArizonaFrameworkSample.

```
[NetworkRpc(Server = true, Channel = NetworkChannelType.ReliableOrdered)]
public void SRPCStep1(NetworkRpcParams inParams = new NetworkRpcParams())
{
    Debug.Log("SRPC Step 1a");

    CRPCStep2(new NetworkRpcParams
    {
        TargetIds = new uint[] { inParams.SenderId }
    });

    Debug.Log("SRPC Step 1b");
}

[NetworkRpc(Client = true, Channel = NetworkChannelType.ReliableOrdered)]
public void CRPCStep2(NetworkRpcParams inParams = new NetworkRpcParams())
{
    Debug.Log("CRPC Step 2a");

    SRPCStep3();
    // CallingANonRpcFunctionInBetweenDoesNotChangeBehavior();

    Debug.Log("CRPC Step 2b");
}

void CallingANonRpcFunctionInBetweenDoesNotChangeBehavior()
{
    Debug.Log("In between");
    SRPCStep3();
}

[NetworkRpc(Server = true, Channel = NetworkChannelType.ReliableOrdered)]
public void SRPCStep3(NetworkRpcParams inParams = new NetworkRpcParams())
{
    Debug.Log("SRPC Step 3a");

    CRPCStep4(new NetworkRpcParams
    {
        TargetIds = new uint[] { inParams.SenderId }
    });

    Debug.Log("SRPC Step 3b");
}

[NetworkRpc(Client = true, Channel = NetworkChannelType.ReliableOrdered)]
public void CRPCStep4(NetworkRpcParams inParams = new NetworkRpcParams())
{
    Debug.Log("CRPC Step 4");
}
```

 
Startup logs:

```
[ 00:33:22.399 ]: [Info] [NetworkReplicator] Add new object 36db5abf4b2d243892e884975497b5cc:Game.MyPlayerState, parent 1234567899634f618472363254c776af:ArizonaFramework.GameInstance
[ 00:33:22.399 ]: [Info] [NetworkReplicator] Remap object ID=186d0a844ee6628c23647f86d6947881 into object 36db5abf4b2d243892e884975497b5cc:Game.MyPlayerState

[ 00:33:22.409 ]: [Info] [NetworkReplicator] Add new object 15342f184695d37f8dcabc8d8ac9cea0:Game.MyPlayerState, parent 1234567899634f618472363254c776af:ArizonaFramework.GameInstance
[ 00:33:22.409 ]: [Info] [NetworkReplicator] Remap object ID=672ac09649442bc2f48924bd3dd16012 into object 15342f184695d37f8dcabc8d8ac9cea0:Game.MyPlayerState

[ 00:33:22.409 ]: [Info] [NetworkReplicator] Add new object 4f547e2e4eac87a42eb4ec88226ce284:Game.MyPlayerState, parent 1234567899634f618472363254c776af:ArizonaFramework.GameInstance
[ 00:33:22.409 ]: [Info] [NetworkReplicator] Remap object ID=338313134774bd1105707c8d50572186 into object 4f547e2e4eac87a42eb4ec88226ce284:Game.MyPlayerState
```

Logs after calling the RPCs:

```
[ 00:33:26.063 ]: [Info] CRPC Step 2a
[ 00:33:26.063 ]: [Info] CRPC Step 2b
[ 00:33:26.064 ]: [Warning] Unable to find scripting object with ID=672ac09649442bc2f48924bd3dd16012. Required type ArizonaFramework.PlayerState.
[ 00:33:26.064 ]: [Warning] Unable to find scripting object with ID=186d0a844ee6628c23647f86d6947881. Required type ArizonaFramework.PlayerState.
[ 00:33:26.064 ]: [Warning] Unable to find scripting object with ID=338313134774bd1105707c8d50572186. Required type ArizonaFramework.PlayerState.
[ 00:33:26.163 ]: [Info] CRPC Step 4
```

**Flax version:**
master branch

**Comment by envision3d** [comment]
I also discovered that the ArizonaFramework sample project will sporadically throw similar warnings.  This can be observed by adding `NetworkReplicator.EnableLog = true;` to the bottom of `OnEnable()` inside MainMenu.cs.

I'm not sure if these situations have the same root cause or different, but I thought this information may be helpful.

**Comment by envision3d** [comment]
I've also noticed that all entries within the `GameState.Instance.PlayerStates` entries, as well as `GameState.Instance.LocalPlayerStates` are go `null` for at least one frame after the warning is displayed, and thus any code depending on them will be broken if executed in that frame.

**Comment by mafiesto4** [comment]
It looks like Prefabs spawning was clearing ObjectIds remapping table which should be fixed in 04f1c9a59b59a79b1ecd4005c8bfea25afe40a5f.

Reopen if you still got this issue.
